### PR TITLE
Update/1038 freeform block add more drawer

### PR DESCRIPTION
**Pull request by BoardJames**
This pull request adds the ability of toolbar items to have children (which allows menus) and it uses this to implement a basic more-drawer (aka kitchen sink, aka ellipse menu) for the freeform block.
more-drawer

Still to do:
- ~Add arrow pointing to ellipse button as in mockup (https://github.com/WordPress/gutenberg/issues/1038 )~
- ~Fix scrolling behaviour so it does not go offscreen~

### CODE REVIEW

**Comment by EphoxJames** [comment]
I have updated the more-drawer to add the arrow in the mockup and to handle the sticky behaviour better.

When there is sufficient space it displays above.
more-drawer-above

But if you scroll down and it can't fit then it displays below.
more-drawer-below

File: blocks/library/freeform/freeform-block.js [hunk 1]
```diff
@@ -200,6 +250,25 @@ export default class FreeformBlock extends wp.element.Component {
...
 
+	onScroll() {
+		if ( this.toolbarElem ) {
+			let blockControls = findDOMNode( this.toolbarElem );
+			while ( blockControls ) {
+				if ( blockControls.classList.contains( 'editor-visual-editor__block-controls' ) ) {
+					break;
+				}
+				blockControls = blockControls.parentElement;
+			}
```
Comment (youknowriad) [inline blocks/library/freeform/freeform-block.js]: What element are we trying to get here? Could we use `element.closest()` instead of looping like this (We just need to include the polyfill `import 'element-closest';`?

File: blocks/library/freeform/freeform-block.js [hunk 2]
```diff
@@ -287,7 +365,19 @@ export default class FreeformBlock extends wp.element.Component {
...
+						title: __( 'More' ),
+						isActive: showMore,
+						onClick: this.toggleMoreDrawer,
+						children: (
+							showMore && <div className={ moreDrawerClasses }>
+								<div className="more-draw__arrow" />
+								<Toolbar controls={ this.mapControls( MORE_CONTROLS ) } />
+							</div>
+						),
+					} ) }
```
Comment (youknowriad) [inline blocks/library/freeform/freeform-block.js]: Should we consolidate the two `<BlockControls>` usage and the `<Fill name="Formatting.Toolbar">` into a unique `<BlockControls>`?

Something like

```js
<BlockControls>
  <Toolbar />
  <Toolbar />
  <Toolbar />
</BlockControls>
```

File: blocks/library/freeform/freeform-block.js (hunk 1 above)
Comment (EphoxJames) [inline blocks/library/freeform/freeform-block.js]: I considered element.closest() but dismissed it because IE does not support it. I didn't know that the polyfill was available.

File: blocks/library/freeform/freeform-block.js (hunk 2 above)
Comment (EphoxJames) [inline blocks/library/freeform/freeform-block.js]: That would make things much cleaner. It should probably be done as a separate pull request though.

File: blocks/library/freeform/freeform-block.js (hunk 1 above)
Comment (EphoxJames) [inline blocks/library/freeform/freeform-block.js]: I have updated to use the polyfill.

File: blocks/library/freeform/freeform-block.js (hunk 2 above)
Comment (EphoxJames) [inline blocks/library/freeform/freeform-block.js]: Nevermind, I see how it can be done now - I had assumed that the controls prop were the only way to set the content of the <BlockControls> element.
